gossip/gossip.go

```diff
@@ -300,25 +301,23 @@ func (g *Gossip) hasIncoming(addr net.Addr) bool {
 	return g.incoming.hasAddr(addr)
 }
 
-// parseBootstrapAddresses parses the gossip bootstrap addresses
+// initializeBootstrapAddresses parses the gossip bootstrap addresses
 // passed via -gossip command line flag.
-func (g *Gossip) parseBootstrapAddresses() {
-	addresses := strings.Split(g.gossipBootstrap, ",")
-	for _, addr := range addresses {
-		addr = strings.TrimSpace(addr)
-		if len(addr) == 0 {
-			continue
-		}
-		_, err := net.ResolveTCPAddr("tcp", addr)
-		if err != nil {
-			log.Errorf("invalid gossip bootstrap address %s: %s", addr, err)
-			continue
+func (g *Gossip) initializeBootstrapAddresses() {
+	for _, addr := range g.gossipBootstrap {
+		if addr.Network() == "tcp" {
+			_, err := net.ResolveTCPAddr("tcp", addr.String())
+			if err != nil {
+				log.Errorf("invalid gossip bootstrap address %s: %s", addr, err)
+				continue
+			}
 		}
-		g.bootstraps.addAddr(util.MakeRawAddr("tcp", addr))
+
+		g.bootstraps.addAddr(addr)
```

Comment:
Currently we add `addr` in `g.gossipBootstrap` whenever `initializeBootstrapAddresses()` is called. I guess we want to clear `g.bootstraps` before adding `addr`. If that's correct, I'll make a change with a separate PR.